### ISSUE TITLE
Dissconnecting server from a singleplayer world (1.20.1) due to rendering error.

### ISSUE THREAD

**Issue by Lucksinia** [hammertater/treechop#240]
So, yeah, relevant part of the logs here:

```java
[19:15:09] [Render thread/ERROR]: Error rendering overlay 'treechop:chopping_indicator'
java.lang.NullPointerException: null
	at java.util.Comparator.lambda$comparing$77a9974f$1(Comparator.java:473) ~[?:?]
	at java.util.PriorityQueue.siftDownUsingComparator(PriorityQueue.java:712) ~[?:?]
	at java.util.PriorityQueue.poll(PriorityQueue.java:585) ~[?:?]
	at ht.tuber.graph.FloodFillImpl.lambda$fill$2(FloodFillImpl.java:29) ~[TreeChop-1.20.1-forge-0.18.3.jar%23227!/:0.18.3]
	at java.util.stream.ReferencePipeline$15$1.accept(ReferencePipeline.java:540) ~[?:?]
	at java.util.stream.Stream$2.tryAdvance(Stream.java:1539) ~[?:?]
	at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129) ~[?:?]
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230) ~[?:?]
	at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:632) ~[?:?]
	at ht.treechop.common.chop.LazyTreeData.hasLeaves(LazyTreeData.java:99) ~[TreeChop-1.20.1-forge-0.18.3.jar%23227!/:0.18.3]
	at ht.treechop.api.AbstractTreeData.isAProperTree(AbstractTreeData.java:14) ~[TreeChop-1.20.1-forge-0.18.3.jar%23227!/:0.18.3]
	at ht.treechop.client.gui.screen.ChopIndicator.blockCanBeChopped(ChopIndicator.java:79) ~[TreeChop-1.20.1-forge-0.18.3.jar%23227!/:0.18.3]
	at ht.treechop.client.gui.screen.ChopIndicator.render(ChopIndicator.java:36) ~[TreeChop-1.20.1-forge-0.18.3.jar%23227!/:0.18.3]
	at ht.treechop.client.ForgeClient.lambda$onRegisterOverlays$0(ForgeClient.java:42) ~[TreeChop-1.20.1-forge-0.18.3.jar%23227!/:0.18.3]
	at net.minecraftforge.client.gui.overlay.ForgeGui.lambda$render$0(ForgeGui.java:126) ~[forge-1.20.1-47.2.4-universal.jar%23233!/:?]
	at com.google.common.collect.ImmutableList.forEach(ImmutableList.java:422) ~[guava-31.1-jre.jar%23109!/:?]
	at net.minecraftforge.client.gui.overlay.ForgeGui.m_280421_(ForgeGui.java:121) ~[forge-1.20.1-47.2.4-universal.jar%23233!/:?]
	at net.minecraft.client.renderer.GameRenderer.m_109093_(GameRenderer.java:945) ~[client-1.20.1-20230612.114412-srg.jar%23228!/:?]
	at net.minecraft.client.Minecraft.m_91383_(Minecraft.java:1146) ~[client-1.20.1-20230612.114412-srg.jar%23228!/:?]
	at net.minecraft.client.Minecraft.m_91374_(Minecraft.java:718) ~[client-1.20.1-20230612.114412-srg.jar%23228!/:?]
	at net.minecraft.client.main.Main.main(Main.java:218) ~[forge-47.2.4.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.runTarget(CommonLaunchHandler.java:111) ~[fmlloader-1.20.1-47.2.4.jar:?]
	at net.minecraftforge.fml.loading.targets.CommonLaunchHandler.clientService(CommonLaunchHandler.java:99) ~[fmlloader-1.20.1-47.2.4.jar:?]
	at net.minecraftforge.fml.loading.targets.CommonClientLaunchHandler.lambda$makeService$0(CommonClientLaunchHandler.java:25) ~[fmlloader-1.20.1-47.2.4.jar:?]
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:30) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:53) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:71) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:108) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:78) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:26) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.modlauncher.BootstrapLaunchConsumer.accept(BootstrapLaunchConsumer.java:23) ~[modlauncher-10.0.9.jar:?]
	at cpw.mods.bootstraplauncher.BootstrapLauncher.main(BootstrapLauncher.java:141) ~[bootstraplauncher-1.1.2.jar:?]
[19:15:09] [Netty Server IO #1/ERROR]: Received class net.minecraft.network.protocol.game.ClientboundDisconnectPacket that couldn't be processed
java.lang.ClassCastException: class net.minecraft.server.network.ServerGamePacketListenerImpl cannot be cast to class net.minecraft.network.protocol.game.ClientGamePacketListener (net.minecraft.server.network.ServerGamePacketListenerImpl and net.minecraft.network.protocol.game.ClientGamePacketListener are in module minecraft@1.20.1 of loader 'TRANSFORMER' @5d512ddb)
	at net.minecraft.network.protocol.game.ClientboundDisconnectPacket.m_5797_(ClientboundDisconnectPacket.java:7) ~[client-1.20.1-20230612.114412-srg.jar%23228!/:?]
	at net.minecraft.network.Connection.m_129517_(Connection.java:174) ~[client-1.20.1-20230612.114412-srg.jar%23228!/:?]
	at net.minecraft.network.Connection.channelRead0(Connection.java:159) ~[client-1.20.1-20230612.114412-srg.jar%23228!/:?]
	at net.minecraft.network.Connection.channelRead0(Connection.java:50) ~[client-1.20.1-20230612.114412-srg.jar%23228!/:?]
	at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.local.LocalChannel.readInbound(LocalChannel.java:299) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.local.LocalChannel.finishPeerRead0(LocalChannel.java:445) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.local.LocalChannel.access$400(LocalChannel.java:50) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.channel.local.LocalChannel$5.run(LocalChannel.java:403) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174) ~[netty-common-4.1.82.Final.jar%23123!/:4.1.82.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167) ~[netty-common-4.1.82.Final.jar%23123!/:4.1.82.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470) ~[netty-common-4.1.82.Final.jar%23123!/:4.1.82.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569) ~[netty-transport-4.1.82.Final.jar%23128!/:4.1.82.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.82.Final.jar%23123!/:4.1.82.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.82.Final.jar%23123!/:4.1.82.Final]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
[19:15:09] [Server thread/INFO]: Lucksinia lost connection: Server sent an invalid packet
[19:15:09] [Server thread/INFO]: Lucksinia left the game
```

I don't know why that was, but after disabling jar in curseforge(or deleting it from mods entirely), all worked as expected.
I have no idea why, but it's trying to connect to multiplayer server? and then complains about invalid packet and boots me out to the server screen in game.

**Comment by hammertater** [comment]
Sorry about the crashes. It's fixed in 1.19.2, I'll port it to 1.20.1 eventually

**Comment by hammertater** [comment]
Fixed in 0.18.7, sorry for the wait!
